src/Composer/Package/ExtraDownload.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace LastCall\DownloadsPlugin\Composer\Package;
+
+use Composer\Package\Package;
+use Composer\Package\PackageInterface;
+use LastCall\DownloadsPlugin\Enum\Type;
+use LastCall\DownloadsPlugin\Model\Hash;
+
+class ExtraDownload extends Package implements ExtraDownloadInterface
+{
+    private string $downloadedPath;
+    private ?Hash $hash;
+    private PackageInterface $parent;
+    private Type $extraDownloadType;
+    private array $executable;
+    private array $ignore;
+
+    public function setExecutable(array $executable): void
+    {
+        $this->executable = $executable;
+    }
+
+    public function getExecutable(): array
+    {
+        return $this->executable;
+    }
+
+    public function setIgnore(array $ignore): void
+    {
+        $this->ignore = $ignore;
+    }
+
+    public function getIgnore(): array
+    {
+        return $this->ignore;
+    }
+
+    public function setHash(?Hash $hash): void
+    {
+        $this->hash = $hash;
+    }
+
+    public function getHash(): ?Hash
+    {
+        return $this->hash;
+    }
```

Comment:
why would the hash be set here, rather than constructor injected and `validate`'d?

Reply:
At the time this ExtraDownload object is created, the file itself is not downloaded ( `downloadedPath` is not set), so we can't validate the downloaded file against this hash.

To prevent changing the hash, I made it readonly.